internal/host/host_metrics_reporter.go

```diff
@@ -0,0 +1,152 @@
+package host
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/openshift/assisted-service/internal/metrics"
+	"github.com/openshift/assisted-service/models"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/sirupsen/logrus"
+)
+
+type HostMetricsReporterAPI interface {
+	ReportValidationFailedMetrics(ctx context.Context, h *models.Host) error
+}
+
+type HostMetricsReporter struct {
+	log        logrus.FieldLogger
+	metricsApi metrics.API
+}
+
+func NewHostMetricsReporter(log logrus.FieldLogger, metricsApi metrics.API) *HostMetricsReporter {
+	return &HostMetricsReporter{
+		log:        log,
+		metricsApi: metricsApi,
+	}
+}
+
+func (m *HostMetricsReporter) ReportValidationFailedMetrics(ctx context.Context, h *models.Host) error {
+	log := logutil.FromContext(ctx, m.log)
+	if h.ValidationsInfo == "" {
+		log.Warnf("Host %s in cluster %s doesn't contain any validations info, cannot report metrics for that host", h.ID, h.ClusterID)
+		return nil
+	}
+	var validationRes ValidationsStatus
+	if err := json.Unmarshal([]byte(h.ValidationsInfo), &validationRes); err != nil {
+		log.WithError(err).Errorf("Failed to unmarshal validations info from host %s in cluster %s", h.ID, h.ClusterID)
+		return err
+	}
+	for _, vRes := range validationRes {
+		for _, v := range vRes {
+			if v.Status == ValidationFailure {
+				m.metricsApi.HostValidationFailed(models.HostValidationID(v.ID))
+			}
+		}
+	}
+	return nil
+}
+
+func (m *MetricsManager) ReportHostInstallationMetrics(ctx context.Context, clusterVersion string, clusterID strfmt.UUID, emailDomain string, boot *models.Disk,
```

Comment:
This is just been copy-pasted from metricsManager to show the intention of the PR, but it will need to be tweaked slightly (dependencies and stuff) and tested properly, as it has been done with the simpler `ReportValidationFailedMetrics`